src/Symfony/Component/HttpKernel/EventListener/LocaleAwareListener.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\EventListener;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\HttpKernel\Event\FinishRequestEvent;
+use Symfony\Component\HttpKernel\Event\GetResponseEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+use Symfony\Contracts\Service\LocaleAwareInterface;
+use TypeError;
+
+/**
+ * Inject request locale to each services that implements LocaleAwareInterface.
+ *
+ * @author Pierre Bobiet <pierrebobiet@gmail.com>
+ */
+class LocaleAwareListener implements EventSubscriberInterface
+{
+    /**
+     * @var \Symfony\Contracts\Service\LocaleAwareInterface[]
```

Comment:
usually we put it on the constructor's `@param` notation, as the property type can be inferred from that. Also namespace does not need to be included, this interface is already in "use"